tsdb/querier_test.go

```diff
@@ -1338,55 +1301,36 @@ func TestDeletedIterator_WithSeek(t *testing.T) {
 	act := make([]sample, 1000)
 	for i := 0; i < 1000; i++ {
 		act[i].t = int64(i)
-		act[i].v = rand.Float64()
+		act[i].v = float64(i)
 		app.Append(act[i].t, act[i].v)
 	}
 
 	cases := []struct {
-		r tombstones.Intervals
+		r        tombstones.Intervals
+		seek     int64
+		ok       bool
+		seekedTs int64
 	}{
-		{r: tombstones.Intervals{{Mint: 1, Maxt: 20}}},
-		{r: tombstones.Intervals{{Mint: 1, Maxt: 10}, {Mint: 12, Maxt: 20}, {Mint: 21, Maxt: 23}, {Mint: 25, Maxt: 30}}},
-		{r: tombstones.Intervals{{Mint: 1, Maxt: 10}, {Mint: 12, Maxt: 20}, {Mint: 20, Maxt: 30}}},
-		{r: tombstones.Intervals{{Mint: 1, Maxt: 10}, {Mint: 12, Maxt: 23}, {Mint: 25, Maxt: 30}}},
-		{r: tombstones.Intervals{{Mint: 1, Maxt: 23}, {Mint: 12, Maxt: 20}, {Mint: 25, Maxt: 30}}},
-		{r: tombstones.Intervals{{Mint: 1, Maxt: 23}, {Mint: 12, Maxt: 20}, {Mint: 25, Maxt: 3000}}},
-		{r: tombstones.Intervals{{Mint: 0, Maxt: 2000}}},
-		{r: tombstones.Intervals{{Mint: 500, Maxt: 2000}}},
-		{r: tombstones.Intervals{{Mint: 0, Maxt: 200}}},
-		{r: tombstones.Intervals{{Mint: 1000, Maxt: 20000}}},
+		{r: tombstones.Intervals{{Mint: 1, Maxt: 20}}, seek: 1, ok: true, seekedTs: 21},
```

Comment:
hm , so you seek ts 1 which exists and the result is ts 21? I don't get it.

Reply:
Because `[1, 20]` were deleted. Tombstones are inclusive on both ends.